portfolio/src/main/webapp/script.js

```diff
@@ -33,20 +33,26 @@ function addRandomGreeting() {
   greetingContainer.innerText = greeting;
 }
 
-// Fetch header for each page.
-fetch("templates/header.html")
-  .then(response => response.text())
-  .then(text => {
-    let header = document.createElement("header");
-    header.innerHTML = text;
-    document.querySelector("body").before(header);
-  });
-
-// Fetch footer for each page
-fetch("templates/footer.html")
-  .then(response => response.text())
-  .then(text => {
-    let footer = document.createElement("footer");
-    footer.innerHTML = text;
-    document.querySelector("body").after(footer);
-  });
\ No newline at end of file
+/**
+ * Loads the HTML retrieved from `src` url and inserts it at the `position` relative to `selector`
+ * Example: loadHTML("templates/header.html", "body", "beforebegin") inserts the header before <body>
+ */
+function loadHTML(src, selector, position) {
+  fetch(src)
+    .then(response => response.text())
+    .then(text => {
+      document.querySelector(selector).insertAdjacentHTML(position, text);
+    });
+}
+ 
+/**
+ * Fetches and loads header and footer content for each page.
+ */
+function onBodyLoad() {
+  // Fetch & load header
+  loadHTML("templates/header.html", "body", "beforebegin");
```

Comment:
Not required, but I image the positions of 'beforebegin' and 'afterend' are from a specific set of positions. If this were in google3 I would recommend creating a typescript enum for these values. Since this is in javascript perhaps creating a javascript object that predefines the options here would be useful in preventing typos and unexpected behavior.

Just wanted to point out the concept, You don't have to make a change here.

Reply:
Great, thanks for pointing that out. I'll likely include that in the next PR.